Sources/StreamChat/WebSocketClient/Events/ChannelEvents.swift

```diff
@@ -4,60 +4,173 @@
 
 import Foundation
 
-public struct ChannelUpdatedEvent: ChannelSpecificEvent {
-    public let cid: ChannelId
+public struct ChannelUpdatedEvent: Event {
+    public let channel: ChatChannel
+    public let message: ChatMessage?
+    public let createdAt: Date
+}
 
-    let payload: Any
+extension ChannelUpdatedEvent {
+    struct DTO: ChannelSpecificEvent {
+        let cid: ChannelId
+        let messageId: MessageId?
+        let createdAt: Date
+        let payload: Any
+        
+        init(from response: EventPayload) throws {
+            cid = try response.value(at: \.channel?.cid)
+            createdAt = try response.value(at: \.createdAt)
+            messageId = try? response.value(at: \.message?.id)
+            payload = response
+        }
+    }
     
-    init(from response: EventPayload) throws {
-        cid = try response.value(at: \.cid)
-        payload = response
+    init?(event: DTO, session: DatabaseSession) {
+        guard let channel = session.channel(cid: event.cid) else { return nil }
+                    
+        self.init(
+            channel: channel.asModel(),
+            message: event.messageId.flatMap { session.message(id: $0)?.asModel() },
+            createdAt: event.createdAt
+        )
     }
 }
 
-public struct ChannelDeletedEvent: ChannelSpecificEvent {
-    public let cid: ChannelId
+public struct ChannelDeletedEvent: Event {
+    public let channel: ChatChannel
+    public let user: ChatUser?
```

Comment:
If it's a user who deleted a channel, how about `deletedBy` naming? 🙂

Reply:
Yes I think `deletedBy`makes more sense IMO

Reply:
let's not rename fields, this makes REST docs less + makes it easy to be inconsistent between SDKs

Reply:
Ah I see... yes that also makes sense 👍 If other SDKs are like this, then let's keep it that way. Will resolve my comments 👍